src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -295,11 +295,11 @@ def smooth(self, filter_posterior):
             raise ValueError("Continuous-discrete smoothing is not supported (yet).")
 
         rv_list = self.smooth_list(
-            filter_posterior,
+            filter_posterior.state_rvs,
             filter_posterior.locations,
         )
-        return KalmanPosterior(
-            filter_posterior.locations, rv_list, self, with_smoothing=True
+        return SmoothingPosterior(
+            filter_posterior.locations, rv_list, self, filter_posterior=filter_posterior
```

Comment:
Should we rename these variables to filtering_posterior, too? I don't mind keeping them the way they are, but I wondered what the best practice here is.

Reply:
Good idea